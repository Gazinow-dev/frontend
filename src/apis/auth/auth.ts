import { LoginFetchProps, LoginFetchResponse } from './type';
import mockData from '../../../mocks/loginResponse.json';
import axiosInstance from '../axiosInstance';

export const loginFetch = async (data: LoginFetchProps) => {
  // 추후 서버 연결
  // return await axiosInstance.post<LoginFetchResponse>('/api/v1/member/login', data);
  return mockData;
};
