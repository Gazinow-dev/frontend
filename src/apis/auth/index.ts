import axios, { AxiosError } from 'axios';
import { axiosInstance } from '../axiosInstance';
import type { LoginFormTypes, LoginFetchResponse, LogoutFetchData } from '@/types/apis';
import { API_BASE_URL } from '@env';

export const loginFetch = async (data: LoginFormTypes) => {
  try {
    const res = await axiosInstance.post<{ data: LoginFetchResponse }>(
      '/api/v1/member/login',
      data,
    );
    return res.data.data;
  } catch (err) {
    const er = err as AxiosError;
    throw er;
  }
};

export const tokenReissueFetch = async ({
  accessToken,
  refreshToken,
}: {
  accessToken: string;
  refreshToken: string;
}) => {
  try {
    const res = await axios.post<{ data: LoginFetchResponse }>(
      '/api/v1/member/reissue',
      {
        accessToken,
        refreshToken,
      },
      {
        baseURL: API_BASE_URL,
      },
    );
    return {
      newAccessToken: res.data.data.accessToken,
      newRefreshToken: res.data.data.refreshToken,
    };
  } catch (err) {
    const er = err as AxiosError;
    throw er;
  }
};

export const logoutFetch = async ({ accessToken, refreshToken }: LogoutFetchData) => {
  try {
    await axiosInstance.post(
      '/api/v1/member/logout',
      { accessToken, refreshToken },
    );
  } catch (err) {
    const er = err as AxiosError;
    throw er;
  }
};

export const quitFetch = async () => {
  try {
    await axiosInstance.delete(
      '/api/v1/member/delete_member',
      { data: {} },
    );
  } catch (err) {
    const er = err as AxiosError;
    throw er;
  }
};