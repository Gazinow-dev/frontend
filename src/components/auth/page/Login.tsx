import { Text } from 'react-native';

const Login = () => {
  return <Text>로그인 페이지</Text>;
};

export default Login;
