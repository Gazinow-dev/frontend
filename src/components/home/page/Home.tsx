import { Text } from 'react-native';

const Home = () => {
  return <Text>홈 페이지</Text>;
};

export default Home;
