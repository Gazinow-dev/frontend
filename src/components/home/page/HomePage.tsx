import { Pressable, Text, View } from 'react-native';

import { SEARCH_NAVIGATION, SUBWAY_SEARCH } from '@/constants';
import { useRootNavigation } from '@/navigation/RootNavigation';

const HomePage = () => {
  const rootNavigation = useRootNavigation();

  const navigateSubwaySearch = () => {
    rootNavigation.push(SEARCH_NAVIGATION, { screen: SUBWAY_SEARCH });
  };

  return (
    <View>
      <Text>홈 페이지</Text>
      <Pressable onPress={navigateSubwaySearch}>
        <Text>지하철검색 페이지 이동</Text>
      </Pressable>
    </View>
  );
};

export default HomePage;
