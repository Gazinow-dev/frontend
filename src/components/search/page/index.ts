export { default as SubwaySearchPage } from './SubwaySearchPage';
