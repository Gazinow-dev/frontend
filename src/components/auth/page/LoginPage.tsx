import { useEffect, useState } from 'react';
import { View } from 'react-native';

import { Input } from '@/components/common/atoms';
import { TextButton } from '@/components/common/molecules';
import { useTryLogin } from '@/hooks';
import { LoginFormTypes } from '@/types/apis';

const initialFormState: LoginFormTypes = {
  email: '',
  password: '',
};

const LoginPage = () => {
  const [formData, setFormData] = useState<LoginFormTypes>(initialFormState);

  const tryLogin = useTryLogin();

  const changeFormText = (type: 'email' | 'password', text: string) => {
    setFormData((prev) => ({ ...prev, [type]: text }));
  };

  const submitFormData = () => {
    tryLogin('submit', formData);
  };

  useEffect(() => {
    tryLogin('auto');
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <View>
      <Input
        placeholder="이메일을 입력해주세요"
        value={formData.email}
        onChangeText={(text) => changeFormText('email', text)}
        inputMode="email"
      />
      <Input
        placeholder="비밀번호를 입력해주세요"
        value={formData.password}
        onChangeText={(text) => changeFormText('password', text)}
        secureTextEntry
      />
      <TextButton value="로그인" onPress={submitFormData} />
    </View>
  );
};

export default LoginPage;
