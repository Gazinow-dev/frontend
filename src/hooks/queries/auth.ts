import { useMutation } from 'react-query';

import { loginFetch, autoLoginFetch } from '@/apis/auth';

export const useLoginMutation = () => {
  return useMutation(loginFetch);
};

export const useAutoLoginMutation = () => {
  const { data, mutate: autoLoginFetching } = useMutation(autoLoginFetch);

  return { data, autoLoginFetching };
};
