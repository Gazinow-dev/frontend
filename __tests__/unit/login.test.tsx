import React from 'react';
import { render } from '@testing-library/react-native';
import { describe, expect, it } from '@jest/globals';
import { Login } from '@/components/auth/page';

describe('<Login />', () => {
  it('matches snapshot', () => {
    const screen = render(<Login />);
    const json = screen.toJSON();
    expect(json).toMatchSnapshot();
  });
  it('has input and a button', () => {
    const { getByText, getByPlaceholderText } = render(<Login />);
    getByPlaceholderText('이메일을 입력해주세요');
    getByPlaceholderText('비밀번호를 입력해주세요');
    getByText('로그인');
  });
});
