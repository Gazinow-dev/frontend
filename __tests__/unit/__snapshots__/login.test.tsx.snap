// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> matches snapshot 1`] = `
<Text>
  로그인 페이지
</Text>
`;
